### PR TITLE
Enable Windows build

### PR DESCRIPTION
**Pull request by seanyen**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

* Enabled Windows build.
* Added `seanyen` as a co-maintainer.

### CODE REVIEW

**Comment by seanyen** [comment]
@alexbw this is ready for review and merge. thanks!

**Comment by seanyen** [comment]
cc @jakirkham @blowekamp

**Comment by blowekamp** [comment]
It's be great to use this in my SimpleITK package. I have done something very similar here:
https://github.com/SimpleITK/SimpleITK/blob/master/SuperBuild/lua.cmake

File: recipe/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,67 @@
+cmake_minimum_required(VERSION 3.1)
```
Comment (isuruf) [inline recipe/CMakeLists.txt]: Did you write this yourself or was this taken from somewhere?

**Comment by isuruf** [comment]
Can you also add the license_file entry?

File: recipe/CMakeLists.txt (hunk 1 above)
Comment (seanyen) [reply recipe/CMakeLists.txt]: Originally I took a look at this [one](https://github.com/walterschell/Lua/blob/master/lua-5.3.3/CMakeLists.txt), but I did some modifications to make this version.

Comment (isuruf) [reply recipe/CMakeLists.txt]: Can you add a link at the top to give credit?

**Comment by seanyen** [comment]
> Can you also add the license_file entry?

Sure, btw `Lua` doesn't place a `LICENCE` file by convention, but it bakes it in `doc/readme.html`. That's why I pointed the `license_file` there.

File: recipe/CMakeLists.txt (hunk 1 above)
Comment (seanyen) [reply recipe/CMakeLists.txt]: Sure thing.

**Comment by seanyen** [comment]
@isuruf The test & build is passed, and it is ready to merge. Thanks!

**Pull request merged**